pyupgrade/_plugins/typing_pep563.py

```diff
@@ -0,0 +1,107 @@
+import ast
+import functools
+from collections import deque
+from typing import Iterable
+from typing import List
+from typing import MutableSequence
+from typing import NamedTuple
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+
+from tokenize_rt import Offset
+
+from pyupgrade._ast_helpers import ast_to_offset
+from pyupgrade._data import register
+from pyupgrade._data import State
+from pyupgrade._data import Token
+from pyupgrade._data import TokenFunc
+
+
+class Replacement(NamedTuple):
+    node: ast.expr
+    new: str
+
+
+def _supported_version(state: State) -> bool:
+    return (
+        state.settings.min_version >= (3, 10) or (
+            'annotations' in state.from_imports['__future__']
+        )
+    )
+
+
+def _dequote(i: int, tokens: List[Token], *, new: str) -> None:
+    tokens[i] = tokens[i]._replace(src=new)
+
+
+def _replace_string_literal(
+    annotation: Optional[ast.expr],
+) -> Sequence[Replacement]:
+    nodes: MutableSequence[ast.AST] = deque()
+    if annotation is not None:
+        nodes.append(annotation)
+    replacements: MutableSequence[Replacement] = deque()
+
+    while nodes:
+        node = nodes.pop()
+        if isinstance(node, ast.Subscript):
+            if isinstance(node.value, ast.Name):
+                if node.value.id not in {
+                    'Literal',
+                    'NamedTuple',
+                    'TypedDict',
+                    'Annotated',
```

Comment:
we should probably handle these as well from `mypy-extensions` (even though they haven't made it into standard typing yet) -- don't want to break people's code without an escape hatch: https://github.com/python/mypy_extensions/blob/dffd016aeb20851c6059b0077be033611f174500/mypy_extensions.py#L109-L136